src/continuous_agg.c

```diff
@@ -400,21 +294,60 @@ ts_continuous_agg_find_by_mat_hypertable_id(int32 mat_hypertable_id)
 }
 
 ContinuousAgg *
-ts_continuous_agg_find_by_view_name(const char *schema, const char *name)
+ts_continuous_agg_find_by_view_name(const char *schema, const char *name,
+									ContinuousAggViewType type)
 {
-	ScanIterator iterator =
-		ts_scan_iterator_create(CONTINUOUS_AGG, AccessShareLock, CurrentMemoryContext);
+	ScanIterator iterator;
 	ContinuousAgg *ca = NULL;
+	AttrNumber view_name_attrnum = 0;
+	AttrNumber schema_name_attrnum = 0;
 	int count = 0;
 
+	Assert(schema);
+	Assert(name);
+
+	switch (type)
+	{
+		case ContinuousAggUserView:
+			schema_name_attrnum = Anum_continuous_agg_user_view_schema;
+			view_name_attrnum = Anum_continuous_agg_user_view_name;
+			break;
+		case ContinuousAggPartialView:
+			schema_name_attrnum = Anum_continuous_agg_partial_view_schema;
+			view_name_attrnum = Anum_continuous_agg_partial_view_name;
+			break;
+		case ContinuousAggDirectView:
+			schema_name_attrnum = Anum_continuous_agg_direct_view_schema;
+			view_name_attrnum = Anum_continuous_agg_direct_view_name;
+			break;
```

Comment:
These cases are not tested. Are they possible? Should there be test covering this for partial view? I guess tests can be done outside this PR.